bittensor/chain_data.py

```diff
@@ -264,15 +264,43 @@ def from_neuron_info(cls, neuron_info: dict) -> "AxonInfo":
             coldkey=neuron_info["coldkey"],
         )
 
-    def to_parameter_dict(self) -> dict[str, Union[int, str]]:
-        r"""Returns a dict of the subnet info."""
+    def _to_parameter_dict_torch(self) -> "torch.nn.ParameterDict":
```

Comment:
You could also do something like this:

```python

def _to_parameter_dict(self, output_type: str) -> Union["torch.nn.ParameterDict", Dict[str, Union[int, str]]]:
    """Returns a dictionary of the subnet info in the specified format."""
    if output_type == 'torch':
        return torch.nn.ParameterDict(self.__dict__)
    elif output_type == 'numpy':
        return self.__dict__
    else:
        raise ValueError(f"Unsupported output_type: {output_type}. Supported types are 'torch' and 'numpy'.")
```

Reply:
This is probably a better approach. Thanks.